File: Best Time to Buy and Sell Stock II/Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        if(prices.empty())  return 0;
        if(prices.size() == 1)  return 0;
        int profit = 0;
        for(unsigned int i = 0; i < prices.size() - 1; ++i){
            if(prices[i + 1] > prices[i])
                profit += prices[i + 1] - prices[i];
        }
        return profit;
    }
};

File: Best Time to Buy and Sell Stock II/Best Time to Buy and Sell Stock II.py
class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        profit = 0
        for i in range(len(prices)):
            if(i == len(prices) - 1):
                continue
            if(prices[i] < prices[i + 1]):
                profit += prices[i + 1] - prices[i]
        return profit
